src/main/java/wheresmymoney/ExpenseList.java

```diff
@@ -0,0 +1,172 @@
+     * @param category New category of expense
+     */
+    public void addExpense(Float price, String description, String category) {
+        Logging.log(Level.INFO,
+                String.format("Adding expense with parameters: %f, %s, %s", price, description, category));
+        Expense expense = new Expense(price, description, category);
+        assert (expense != null);
+        expenses.add(expense);
+        Logging.log(Level.INFO, "Successfully added expense.");
+    }
+
+    /**
+     * Edit the details of an expense given its position in the list
+     *
+     * @param index index of Expense in the list that is to be edited
+     * @param category New category of expense
+     * @param price New price of expense
+     * @param description New description of expense
+     */
+    public void editExpense(int index, Float price, String description, String category) throws WheresMyMoneyException {
+        try {
+            Logging.log(Level.INFO, "Attempting to edit expense.");
+            Expense expense = expenses.get(index);
+            assert (expense != null);
+            expense.setPrice(price);
+            expense.setDescription(description);
+            expense.setCategory(category);
+            Logging.log(Level.INFO, "Successfully edited expense.");
+        } catch (WheresMyMoneyException e) {
+            Logging.log(Level.INFO, "Failure when editing expense.");
+            throw new WheresMyMoneyException(e.getMessage());
+        }
+    }
+
+    public void deleteExpense(int index) throws WheresMyMoneyException {
+        if (index < 0 || index >= expenses.size()) {
+            throw new WheresMyMoneyException("Index out of range!");
+        }
+        expenses.remove(index);
+    }
+
+    /**
+     * Returns the list of all expenses from the specified category
+     *
+     * @param category Category of expense
+     */
+    public ArrayList<Expense> listByCategory(String category) {
+        ArrayList<Expense> expensesFromCategory = new ArrayList<>();
+        for (Expense expense: expenses) {
+            if (expense.category.equals(category)) {
+                Logging.log(Level.INFO, "Found matching expense: " + expense.description);
+                expensesFromCategory.add(expense);
+            }
+        }
+        return expensesFromCategory;
+    }
+
+    /**
+     * Loads from a csv file into Expense List.
+     *
+     * @param filePath File Path to read csv
+     */
+    public void loadFromCsv(String filePath) throws StorageException {
+        try {
+            File file = new File(filePath);
+            FileReader reader = new FileReader(file);
+            CSVReader csvReader = new CSVReader(reader);
+
+            csvReader.readNext(); // Skip the header
+            String[] line;
+            while ((line = csvReader.readNext()) != null) {
+                addExpense(Float.parseFloat(line[2]), line[1], line[0]);
+            }
+
+            // closing writer connection
```

Comment:
It's a redundant comment and also looks incorrect.
Comment mentions writer; code references reader!